packages/arb-avm-cpp/data_storage/src/value/code.cpp

```diff
@@ -45,84 +50,152 @@ std::array<unsigned char, segment_key_size> segment_key(uint64_t segment_id) {
     return key;
 }
 
-struct RawCodePoint {
+std::array<unsigned char, segment_chunk_key_size> segment_op_chunk_key_raw(
+    uint64_t segment_id,
+    uint64_t chunk) {
+    std::array<unsigned char, segment_chunk_key_size> key{};
+    auto it = std::copy(segment_key_prefix.begin(), segment_key_prefix.end(),
+                        key.begin());
+
+    auto big_id = boost::endian::native_to_big(segment_id);
+    auto big_id_ptr = reinterpret_cast<const char*>(&big_id);
+    it = std::copy(big_id_ptr, big_id_ptr + sizeof(big_id), it);
+
+    auto big_chunk = boost::endian::native_to_big(chunk);
+    auto big_chunk_ptr = reinterpret_cast<const char*>(&big_chunk);
+    std::copy(big_chunk_ptr, big_chunk_ptr + sizeof(big_chunk), it);
+    key.back() = 0;
```

Comment:
Shouldn't this be `key.push_back(0);` or similar, instead of overwriting a byte of big_chunk? That seems dangerous given that it's big endian.

Reply:
key is an array rather than a vector so this should just set the last byte which is uninitialized before this

Reply:
Ah, I missed that, thanks. Could we add an assert before this that `it - key == segment_key_size - 1`?